client/firmwareOSource_service.go

```diff
@@ -57,6 +57,7 @@ func (sm *ServiceManager) CreateOSource(name string, description string, firmwar
 	rn := fmt.Sprintf("osrc-%s", name)
 	parentDn := fmt.Sprintf("uni/fabric/fwrepop")
 	firmwareOSource := models.NewOSource(rn, parentDn, description, firmwareOSourceAttr)
+	firmwareOSource.Status = ""  // this avoids API failure in case MO is already created in APIC
```

Comment:
Why is this line necessary if NewOSource already returns a struct with Status set to empty string?  If its not needed  - remove it.

Reply:
Will take care of it.